### ISSUE TITLE
Rule check-param-names does not recognize optional parameters.

### ISSUE THREAD

**Issue by bennieswart** [gajus/eslint-plugin-jsdoc#33]
Consider the following code:
```js
/**
 * When the Edit Invoice button is clicked.
 * @param {invoice} invoice - The selected Invoice.
 * @param {string} [subscreen]- Subscreen the BillingScreen should load on when opened.
 */
AccountScreen.prototype.onViewEditInvoiceClick = function (invoice, subscreen) {
    // ...
};
```
The rule check-param-names gives this error:
error  Expected <span>@</span>param names to be "invoice, subscreen". Got "invoice, [subscreen=false]-"  jsdoc/check-param-names

To my understanding this should not be an error. The same goes for [subscreen=somedefault], which also reports an error.

Is my understanding correct? Will someone fix this? I'm willing, but will have to get familiarized with the code first.

Edit: None of which I have said is true. There was a missing space between the `[subscreen]` and the dash, which caused all the trouble. It seems that optional parameters are indeed handled correctly.
